### PR TITLE
chore: added basic example for validation

### PR DESCRIPTION
**Pull request by brenkao**
same as title

### CODE REVIEW

File: examples/validator.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        ..., description="The steps to perform the task or answer the question."
+    )
+
+
+try:
+    content = chat.create(QuestionPrompt(question="cake recipe?"))
+    guide = chat.extract(HowToGuide, str(content))
+    print(guide)
+except ValidationError as e:
+    print(e)
```
Comment (willbakst) [inline examples/validator.py:75]: I think the only thing missing is showing how to handle the validation errors when it isn't a question or is off topic (i.e. response with hard-coded "That's not a question" or something)

File: examples/validator.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        ..., description="The steps to perform the task or answer the question."
+    )
+
+
+try:
+    content = chat.create(QuestionPrompt(question="cake recipe?"))
+    guide = chat.extract(HowToGuide, str(content))
+    print(guide)
+except ValidationError:
+    print("Either the text was not a question or it is not relevant to the topic.")
```
Comment (willbakst) [inline examples/validator.py:75]: I was thinking more like:

```python
except ValidationError as e:
    print(e)
    #> The text is not a question: cake recipe
```

Comment (brenkao) [reply examples/validator.py:75]: ah

File: examples/validator.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    topic: str = topic
+    question: str = Field(..., description="The user's question.")
+
+    call_params = OpenAICallParams(model="gpt-3.5-turbo", temperature=0.1)
+
+    @field_validator("question", mode="after")
+    @classmethod
+    def validate_question(cls, question: str) -> str:
+        class Text(BaseModel):
```
Comment (willbakst) [inline examples/validator.py:42]: marge into `ValidationInfo` and do single extract

Comment (willbakst) [reply examples/validator.py:42]: Follow-up:

What if we showed hitting a "fine-tuned" local model for validation before sending the request to API to scrub it of PII

**Review by willbakst — APPROVED** [review]

**Pull request merged**